tool/pftool.cc

```diff
@@ -785,6 +785,36 @@ void ldmx_daq( const std::string& cmd, PolarfireTarget* pft ) {
     }
     fclose(f);
   }
+  if (cmd=="SCAN"){
+    std::string pagename=BaseMenu::readline("Sub-block name :  ");
+    std::string valuename=BaseMenu::readline("Value name :  ");
+    int iroc=BaseMenu::readline_int("Which ROC :  ");
+    int minvalue=BaseMenu::readline_int("Minimum value :  ");
+    int maxvalue=BaseMenu::readline_int("Maximum value :  ");
+    int step=BaseMenu::readline_int("Step :  ");
+    int nevents=BaseMenu::readline_int("Events per step :  ", 10);
+    std::string fname=BaseMenu::readline("Filename :  ");
+
+    FILE* f=fopen(fname.c_str(),"w");
+    pft->prepareNewRun();
+
+    for(int value = minvalue; value <= maxvalue; value += step){
+      //Catch errors from using incorrect page or value names
+      try{
+        pft->hcal->roc(iroc).applyParameter(pagename, valuename, value);
+
+        for (int ievt=0; ievt<nevents; ievt++) {
+          pft->backend->fc_sendL1A();
+          std::vector<uint32_t> event = pft->daqReadEvent();
+          fwrite(&(event[0]),sizeof(uint32_t),event.size(),f);
+        }
+
+      } catch (const pflib::Exception& e) {
```

Comment:
Do you want to continue attempting different parameters after one of them failed? I would have assumed that this `try-catch` would be outside the `for` loop, but maybe I'm not thinking of a use case.